synapse/util/ratelimitutils.py

```diff
@@ -52,6 +53,34 @@ def new_limiter() -> "_PerHostRatelimiter":
             str, "_PerHostRatelimiter"
         ] = collections.defaultdict(new_limiter)
 
+        # We track the number of affected hosts per time-period so we can
+        # differentiate one really noisy homeserver from a general
+        # ratelimit tuning problem across the federation.
+        LaterGauge(
+            "synapse_rate_limit_sleep_affected_hosts",
+            "Number of hosts that had requests put to sleep",
+            [],
+            lambda: count(
+                bool,
+                [
+                    ratelimiter.should_sleep()
+                    for ratelimiter in self.ratelimiters.values()
+                ],
+            ),
```

Comment:
Though this feels clunky.
I would probably write
`lambda: sum(ratelimiter.should_sleep() for ratelimiter in self.ratelimiters.values())`
No point building an intermediate list (which is what `[... for .. in ...]` does — the above should just be an iterator and moreover it uses the standard `sum` function

Reply:
👍 Good shout

Related to https://github.com/matrix-org/synapse/pull/13541#discussion_r948360863